src/main/java/com/google/sps/servlets/StudyServlet.java

```diff
@@ -0,0 +1,200 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.TransactionOptions.Builder;
+import com.google.appengine.api.datastore.Transaction;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import com.google.gson.Gson;
+
+import com.google.sps.data.Card;
+import com.google.sps.tool.WordSearch;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Comparator;
+import org.joda.time.DateTimeUtils;
+
+/**
+ * Handles the study mode questions and responses by adjusting and sorting a card's based on their
+ * familiarity score and last time seen.
+ */
+@WebServlet("/study")
+public class StudyServlet extends HttpServlet {
+
+  public class QuizCard {
+    private String quizWord;
+    private String cardKey;
+    private ArrayList<String> possibleResponses = new ArrayList<>();
+    private String correctAnswer;
+
+    public QuizCard(Card card) {
+      this.quizWord = card.getTextTranslated();
+      this.cardKey = card.getCardKey();
+      this.correctAnswer = card.getRawText().toLowerCase();
+      this.possibleResponses = WordSearch.generateWordOptions(this.correctAnswer);
+    }
+  }
+
+  private int numOfCardsPerRound = 1;
```

Comment:
const

Reply:
No longer a constant because it can be changed by the servlets methods for testing.

Reply:
done.